asdf/extension/_legacy.py

```diff
@@ -248,11 +258,35 @@ def resolver(self):
         return self.extension_list.resolver
 
 
-default_extensions = _DefaultExtensions()
+_default_extensions = _DefaultExtensions()
+
+
+def __getattr__(name):
+    if name == "default_extensions":
+        warnings.warn(
+            "default_extensions is deprecated. "
+            "Please see the new extension API "
+            "https://asdf.readthedocs.io/en/stable/asdf/extending/converters.html",
+            AsdfDeprecationWarning,
+        )
+        return _default_extensions
+    msg = f"module {__name__!r} has no attribute {name!r}"
+    raise AttributeError(msg)
 
 
 def get_default_resolver():
     """
     Get the resolver that includes mappings from all installed extensions.
     """
-    return default_extensions.resolver
+    warnings.warn(
+        "get_default_resolver is deprecated. "
+        "Please see the new extension API "
+        "https://asdf.readthedocs.io/en/stable/asdf/extending/converters.html",
+        AsdfDeprecationWarning,
+    )
+
+    return _get_default_resolver()
```

Comment:
Same idea as for `get_cached_asdf_extension_list`, I suggest moving this directly into the `asdf.extension.__init__`.